plasmapy/dispersion/two_fluid_dispersion.py

```diff
@@ -0,0 +1,352 @@
+               [0.01534..., 0.01558...]] rad / s>
+    """
+
+    # validate argument ion
+    if not isinstance(ion, Particle):
+        try:
+            ion = Particle(ion)
+        except TypeError:
+            raise TypeError(
+                f"For argument 'ion' expected type {Particle} but got {type(ion)}."
+            )
+    if not (ion.is_ion or ion.is_category("element")):
+        raise ValueError(f"The particle passed for 'ion' must be an ion or element.")
+
+    # validate z_mean
+    if z_mean is None:
+        try:
+            z_mean = abs(ion.integer_charge)
+        except ChargeError:
+            z_mean = 1
+    else:
+        if not isinstance(z_mean, (int, np.integer, float, np.floating)):
+            raise TypeError(
+                f"Expected int or float for argument 'z_mean', but got {type(z_mean)}."
+            )
+        z_mean = abs(z_mean)
+
+    # validate arguments
+    for arg_name in ("B", "n_i", "T_e", "T_i"):
+        val = locals()[arg_name].squeeze()
+        if val.shape != ():
+            raise ValueError(
+                f"Argument '{arg_name}' must a single value and not an array of "
+                f"shape {val.shape}."
+            )
+        locals()[arg_name] = val
+
+    # validate arguments
+    for arg_name in ("gamma_e", "gamma_i"):
+        if not isinstance(locals()[arg_name], (int, np.integer, float, np.floating)):
+            raise TypeError(
+                f"Expected int or float for argument '{arg_name}', but got "
+                f"{type(locals()[arg_name])}."
+            )
+
+    # validate argument k
+    k = k.squeeze()
+    if not (k.ndim == 0 or k.ndim == 1):
+        raise ValueError(
+            f"Argument 'k' needs to be a single valued or 1D array astropy Quantity,"
+            f" got array of shape {k.shape}."
+        )
+    if np.any(k <= 0):
+        raise ValueError(f"Argument 'k' can not be a or have negative values.")
+
+    # validate argument theta
+    theta = theta.squeeze()
+    theta = theta.to(u.radian)
+    if not (theta.ndim == 0 or theta.ndim == 1):
+        raise ValueError(
+            f"Argument 'theta' needs to be a single valued or 1D array astropy "
+            f"Quantity, got array of shape {k.shape}."
+        )
+
+    # Calc needed plasma parameters
+    n_e = z_mean * n_i
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=PhysicsWarning)
+        c_s = pfp.ion_sound_speed(
+            T_e=T_e,
+            T_i=T_i,
+            ion=ion,
+            n_e=n_e,
+            gamma_e=gamma_e,
+            gamma_i=gamma_i,
+            z_mean=z_mean,
+        )
+    v_A = pfp.Alfven_speed(B, n_i, ion=ion, z_mean=z_mean)
+    omega_ci = pfp.gyrofrequency(B=B, particle=ion, signed=False, Z=z_mean)
+    omega_pe = pfp.plasma_frequency(n=n_e, particle="e-")
+
+    # Bellan2012JGR params equation 32
+    alpha = np.cos(theta.value) ** 2
+    beta = (c_s / v_A).to(u.dimensionless_unscaled).value ** 2
```

Comment:
If I remember correctly this isn't strictly the standard plasma beta (gas pressure to magnetic pressure ratio), but rather that there's a factor of order unity that's dependent on what `gamma_i` is (I think).  I'm wondering if we would want to call this `beta_prime`.

Reply:
Unless I'm missing something, I believe this is the plasma beta.  Bellan and Stringer use the beta nomenclature, so I think it's better to keep `beta` to be consistent with the literature, and the docstrings.

Reply:
I agree with @rocco8773, for the sake of consistency with the original document, let's just leave it as `beta`. Even in literature we always find two different definitions of `beta` which differ from each other by a factor of 2, so there is that!

Reply:
Correct.  The Bellan beta is a factor of 2 off from the typical beta, `beta = nkT/(B/(2 mu0)) = 2 beta_bellan`.